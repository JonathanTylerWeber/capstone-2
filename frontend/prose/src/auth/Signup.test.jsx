import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import SignupForm from './Signup';

describe('SignupForm Component', () => {
  const mockHandleSignup = vi.fn();

  beforeEach(() => {
    mockHandleSignup.mockClear();
  });

  it("matches snapshot", function () {
    const { asFragment } = render(<SignupForm />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the signup form correctly', () => {
    const { getByLabelText, getByPlaceholderText } = render(<SignupForm handleSignup={mockHandleSignup} />);

    getByLabelText(/Username:/i);
    getByPlaceholderText(/Enter username/i);
    getByLabelText(/Email:/i);
    getByPlaceholderText(/Enter email/i);
    getByLabelText(/Password:/i);
    getByPlaceholderText(/Enter password/i);
  });

  it('calls handleSignup with form data when submitted', async () => {
    const { getByLabelText, getByText } = render(<SignupForm handleSignup={mockHandleSignup} />);

    fireEvent.change(getByLabelText(/Username:/i), { target: { value: 'testuser' } });
    fireEvent.change(getByLabelText(/Email:/i), { target: { value: 'test@example.com' } });
    fireEvent.change(getByLabelText(/Password:/i), { target: { value: 'password123' } });

    fireEvent.submit(getByText(/Submit/i));

    await waitFor(() => {
      expect(mockHandleSignup).toHaveBeenCalledTimes(1);
      expect(mockHandleSignup).toHaveBeenCalledWith({
        username: 'testuser',
        email: 'test@example.com',
        password: 'password123'
      });
    });
  });

});
