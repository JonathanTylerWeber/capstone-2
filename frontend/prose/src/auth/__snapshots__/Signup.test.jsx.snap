// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignupForm Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-auto cntr container"
  >
    <form
      class=""
    >
      <h1
        class="signup-h1"
      >
        Sign Up
      </h1>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="username"
        >
          Username:
        </label>
        <input
          class="form-control"
          id="username"
          placeholder="Enter username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="email"
        >
          Email:
        </label>
        <input
          class="form-control"
          id="email"
          placeholder="Enter email"
          required=""
          type="email"
          value=""
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="password"
        >
          Password:
        </label>
        <input
          class="form-control"
          id="password"
          placeholder="Enter password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        class="btn auth-btn btn btn-primary"
        type="submit"
      >
        Submit
      </button>
      <a
        class="nav-link my-3 nav-link"
        exact="true"
        href="/login"
      >
        Already have an account?
      </a>
    </form>
  </div>
</DocumentFragment>
`;
